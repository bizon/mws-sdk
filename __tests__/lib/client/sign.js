const sign = require('../../../lib/client/sign')

describe('lib.client.sign', () => {
  it('should generate generate signatures', () => {
    expect(
      sign('secret', {
        method: 'POST',
        domain: 'mws-eu.amazonservices.com',
        path: 'Products/2011-10-01',
        data: {
          Cool: 'Test with spaces'
        }
      })
    ).toMatchSnapshot()

    expect(
      sign('secret', {
        method: 'POST',
        domain: 'mws-eu.amazonservices.com',
        path: 'Products/2011-10-01',
        data: {
          Method: 'Example'
        }
      })
    ).toMatchSnapshot()
  })
})
