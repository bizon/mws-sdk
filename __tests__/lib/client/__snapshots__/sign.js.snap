// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.sign should generate generate signatures 1`] = `"JKuirYHlrTxwQYwG61ULhe+VkyngRdgokEOPSnnxieo="`;

exports[`lib.client.sign should generate generate signatures 2`] = `"nUS2Ev6IZwnNwnVNRWz8hFjJfns5HJLsBrWlMSGwW/A="`;
